src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3485,6 +3486,44 @@ internal void SymbolDeclaredEvent(Symbol symbol)
             EventQueue?.TryEnqueue(new SymbolDeclaredCompilationEvent(this, symbol.GetPublicSymbol()));
         }
 
+        internal override void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder)
+        {
+            WriteValue("checked", Options.CheckOverflow.ToString());
+            WriteValue("nullable", Options.NullableContextOptions.ToString());
+            WriteValue("unsafe", Options.AllowUnsafe.ToString());
+            WriteValue("langversion", LanguageVersion.ToDisplayString());
+
+            var isOptimized = Options.OptimizationLevel == OptimizationLevel.Release ? true : false;
+            WriteValue("optimize", isOptimized.ToString());
+
+            var preprocessorSymbols = GetPreprocessorSymbols();
+            if (preprocessorSymbols.Any())
+            {
+                WriteValue("define", string.Join(",", preprocessorSymbols));
+            }
+
+            void WriteValue(string key, string value)
+            {
+                builder.WriteUTF8(key);
+                builder.WriteByte(0);
+                builder.WriteUTF8(value);
+                builder.WriteByte(0);
+            }
+        }
+
+        private ImmutableArray<string> GetPreprocessorSymbols()
+        {
+            var firstTree = SyntaxTrees
+                .FirstOrDefault(t => t is CSharpSyntaxTree) as CSharpSyntaxTree;
```

Comment:
>FirstOrDefault [](start = 17, length = 14)

Since we are in CSharpCompilation override you can just take the first tree and cast it explicitly to CSharpSyntaxTree - there can be no other trees in the compilation.